src/TrackGenerator.cpp

```diff
@@ -491,6 +528,7 @@ void TrackGenerator::generateTracks() {
     /* Allocate memory for the Tracks */
     try {
       _num_tracks = new int[_num_azim];
+      _num_tracks_by_halfspace = new int[5];
```

Comment:
There are 4 halfspaces and it is easiest to have the index for the beginning and end of each half space stored, so hence 5 values. I created a variable for the number of independent transport sweep loops required, so I'll change this to that value plus 1.